src/storm/mesos/MesosNimbus.java

```diff
@@ -547,29 +547,20 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
 
               String executorDataStr = JSONValue.toJSONString(executorData);
               LOG.info("Launching task with Mesos Executor data: <" + executorDataStr + ">");
+
+              // Create a docker builder if a docker image was defined in the yaml
+              Optional<ContainerInfo.Builder> containerInfoBuilder = createDockerInfoBuilder(getDockerImage());
```

Comment:
Good point. Also I wish some one would refactor assignSlots.....a method this long needs attention.